compiler_opt/distributed/worker.py

```diff
@@ -86,3 +89,21 @@ def get_exception(worker_future: WorkerFuture) -> Optional[Exception]:
     return None
   except Exception as e:  # pylint: disable=broad-except
     return e
+
+
+def get_full_worker_args(worker_class: 'type[Worker]', current_kwargs):
+  """Get the union of given kwargs and gin config.
+
+  This allows the worker hosting process be set up differently from the training
+  process - e.g. no need to initialize gin variables there, for example.
+  """
+  gin_config = {}
+  try:
+    gin_config = gin.get_bindings(worker_class)
+  except:  # pylint: disable=bare-except
+    pass
```

Comment:
what exception can it throw? and shall we silently ignore the exception?

Reply:
left a comment. ValueError, funny enough, because it's not possible to figure out if the object even has a gin config without trying.

Reply:
I'm a bit concerned about the silent ignorance of all exceptions, e.g., spelling mistakes, etc. Ideally we want to separate the exceptions we want to capture and the the exceptions we meant to ignore.

The bottom line here is to capture ValueError instead of ignoring all exceptions

Reply:
Done. I wish they threw something more specific, because "ValueError" is very well all those examples you gave, too (they should be caught by the typechecker, but surely there's others in the same vein)

Maybe I'll contribute a patch to gin.